include/swift/AST/Attr.h

```diff
@@ -1595,7 +1595,8 @@ class DifferentiableAttr final
 
   // Print the attribute to the given stream.
   void print(llvm::raw_ostream &OS, const Decl *D,
-             ModuleDecl *prettyPrintInModule = nullptr) const;
+             AutoDiffParameterIndices *inferredParametersForPrettyPrinting =
```

Comment:
Is it the case that the only thing `print()` will do is to check whether `inferredParametersForPrettyPrinting->parameters.count()` is equal to the actual count so that it will not print it?  In that case, I find passing an `AutoDiffParameterIndices *` argument a bit unnecessary.  It should suffice to just take a Boolean flag (e.g. `omitWRT`) that indicates whether the `wrt:` clause in the attribute should be printed.

Reply:
Done in https://github.com/apple/swift/pull/25324/commits/7759a3db53cb7ad24776de7a7f0b241c8d44781d. Commit description:

```
Use `omitDifferentiationParametersClause` boolean flag.

Caller in TypeCheckProtocol.cpp computes the flag.
Create helper function if there become more callers in Sema.
```

I tried creating a `std::string getDifferentiableAttributeString(const DifferentiableAttr *attr, AbstractFunctionDecl *original)` helper in Sema, but that didn't work because it requires two `AbstractFunctionDecl *` arguments: one for inferring parameters (witness) and one for printing the attribute (requirement). I decided not to create the helper for now.